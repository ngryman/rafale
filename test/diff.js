import test from 'ava'
import { patchesToJSON } from './helpers/to-json'
import { html, diff } from '../'

function macro(t, newVTree, oldVTree) {
  t.snapshot(patchesToJSON(diff(newVTree, oldVTree)))
}

const renderPrimitive = (text) => text
const renderBold = (text) => html`<b>${text}</b>`
const renderItalic = (text) => html`<i>${text}</i>`
const wrap = (content) => html`<p>${content}</p>`

test('morph a root component', macro,
  renderItalic('foo'),
  renderBold('foo')
)

test('morph a component', macro,
  wrap(renderItalic('foo')),
  wrap(renderBold('foo')),
)

test('morph a component into a primitive', macro,
  wrap(renderPrimitive('foo')),
  wrap(renderBold('foo'))
)

test('morph a component into a sequence', macro,
  wrap(['foo', 'bar'].map(renderBold)),
  wrap(renderBold('foo'))
)

test('morph a sequence', macro,
  wrap(['foo', 'bar'].map(renderItalic)),
  wrap(['foo', 'bar'].map(renderBold))
)

test('morph a sequence into a component', macro,
  wrap(renderBold('foo')),
  wrap(['foo', 'bar'].map(renderBold))
)

test('morph a sequence into a primitive', macro,
  wrap(renderPrimitive('foo')),
  wrap(['foo', 'bar'].map(renderBold))
)

test('add items to a sequence', macro,
  wrap(['foo', 'bar', 'baz', 'qux'].map(renderBold)),
  wrap(['foo', 'bar'].map(renderBold))
)

test('remove items from a sequence', macro,
  wrap(['foo', 'baz'].map(renderBold)),
  wrap(['foo', 'bar', 'baz', 'qux'].map(renderBold))
)

test('clear a sequence', macro,
  wrap([].map(renderBold)),
  wrap(['foo', 'bar'].map(renderBold))
)

test('update items of a sequence', macro,
  wrap(['baz', 'qux'].map(renderBold)),
  wrap(['foo', 'bar'].map(renderBold))
)

test('reorder items of a sequence', macro,
  wrap(['qux', 'baz', 'bar', 'foo'].map(renderBold)),
  wrap(['foo', 'bar', 'baz', 'qux'].map(renderBold))
)

test('copy ref even if vnodes are the same', t => {
  const oldVTree = wrap(renderBold('foo'))
  const newVTree = wrap(renderBold('foo'))
  diff(newVTree, oldVTree)
  t.is(newVTree.element, oldVTree.element)
  t.is(newVTree.children[0].element, oldVTree.children[0].element)
})
