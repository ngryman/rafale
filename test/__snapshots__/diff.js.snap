// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add items to a sequence 1`] = `
Array [
  Object {
    "newNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "baz",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "foo",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "bar",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 1,
      "value": null,
    },
    "operation": 7,
  },
  Object {
    "newNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "qux",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "foo",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "bar",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 1,
      "value": null,
    },
    "operation": 7,
  },
]
`;

exports[`clear a sequence 1`] = `
Array [
  Object {
    "newNode": Object {
      "children": Array [],
      "component": null,
      "type": 1,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "foo",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "bar",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 1,
      "value": null,
    },
    "operation": 9,
  },
]
`;

exports[`morph a component 1`] = `
Array [
  Object {
    "newNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<i></i>",
      },
      "type": 2,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "operation": 5,
  },
  Object {
    "newNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<i></i>",
      },
      "type": 2,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "operation": 2,
  },
]
`;

exports[`morph a component into a primitive 1`] = `
Array [
  Object {
    "newNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "foo",
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "operation": 5,
  },
  Object {
    "newNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "foo",
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "operation": 0,
  },
]
`;

exports[`morph a component into a sequence 1`] = `
Array [
  Object {
    "newNode": Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "foo",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "bar",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 1,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "operation": 5,
  },
  Object {
    "newNode": Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "foo",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "bar",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 1,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "operation": 1,
  },
]
`;

exports[`morph a root component 1`] = `
Array [
  Object {
    "newNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<i></i>",
      },
      "type": 2,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "operation": 5,
  },
  Object {
    "newNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<i></i>",
      },
      "type": 2,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "operation": 2,
  },
]
`;

exports[`morph a sequence 1`] = `
Array [
  Object {
    "newNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<i></i>",
      },
      "type": 2,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "operation": 5,
  },
  Object {
    "newNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<i></i>",
      },
      "type": 2,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "operation": 2,
  },
  Object {
    "newNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "bar",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<i></i>",
      },
      "type": 2,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "bar",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "operation": 5,
  },
  Object {
    "newNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "bar",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<i></i>",
      },
      "type": 2,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "bar",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "operation": 2,
  },
]
`;

exports[`morph a sequence into a component 1`] = `
Array [
  Object {
    "newNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "foo",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "bar",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 1,
      "value": null,
    },
    "operation": 4,
  },
  Object {
    "newNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "foo",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "foo",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "bar",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 1,
      "value": null,
    },
    "operation": 2,
  },
]
`;

exports[`morph a sequence into a primitive 1`] = `
Array [
  Object {
    "newNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "foo",
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "foo",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "bar",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 1,
      "value": null,
    },
    "operation": 4,
  },
  Object {
    "newNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "foo",
    },
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "foo",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "bar",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 1,
      "value": null,
    },
    "operation": 0,
  },
]
`;

exports[`remove items from a sequence 1`] = `
Array [
  Object {
    "newNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "baz",
    },
    "oldNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "bar",
    },
    "operation": 6,
  },
  Object {
    "newNode": null,
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "baz",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "operation": 8,
  },
  Object {
    "newNode": null,
    "oldNode": Object {
      "children": Array [
        Object {
          "children": Array [],
          "component": null,
          "type": 0,
          "value": "qux",
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 2,
      "value": null,
    },
    "operation": 8,
  },
]
`;

exports[`reorder items of a sequence 1`] = `
Array [
  Object {
    "newNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "qux",
    },
    "oldNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "foo",
    },
    "operation": 6,
  },
  Object {
    "newNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "baz",
    },
    "oldNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "bar",
    },
    "operation": 6,
  },
  Object {
    "newNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "bar",
    },
    "oldNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "baz",
    },
    "operation": 6,
  },
  Object {
    "newNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "foo",
    },
    "oldNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "qux",
    },
    "operation": 6,
  },
]
`;

exports[`update items of a sequence 1`] = `
Array [
  Object {
    "newNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "baz",
    },
    "oldNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "foo",
    },
    "operation": 6,
  },
  Object {
    "newNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "qux",
    },
    "oldNode": Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "bar",
    },
    "operation": 6,
  },
]
`;
